### PR TITLE
feat: Add hny-network-agent chart

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?
Adds a helm chart for the new Honeycomb Network Agent.

- Closes https://github.com/honeycombio/honeycomb-network-agent/issues/87

## Short description of the changes
- Adds new chart, values, and templated resources

## How to verify that this has the expected result
The new agent can be installed with this chart.

### CODE REVIEW

**Comment by MikeGoldsmith** [comment]
I believe the test is failing because the chart fails to install if neither honeycomb.apikey or an honeycomb.existingSecret is set.

File: charts/hny-network-agent/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  / // /  / |/ / \ \/ /       / |/ / ___  / /_ _    __ ___   ____  / /__       / _ |  ___ _ ___   ___  / /_
+ / _  /  /    /   \  /       /    / / -_)/ __/| |/|/ // _ \ / __/ /  '_/      / __ | / _ `// -_) / _ \/ __/
+/_//_/  /_/|_/    /_/       /_/|_/  \__/ \__/ |__,__/ \___//_/   /_/\_\      /_/ |_| \_, / \__/ /_//_/\__/ 
+                                                                                    /___/                  
+
+
+Honeycomb Network Agent is setup and configured to collect events from your Kubernetes cluster.
+You should see data flowing within a few minutes at https://ui.honeycomb.io.
+
+{{ if .Values.honeycomb.apiKey }}
```
Comment (MikeGoldsmith) [inline charts/hny-network-agent/templates/NOTES.txt:12]: @TylerHelmuth I think we've lost our validation rules for checking the following conditions:
- Neither honeycomb.apiKey or honeycomb.existingSecret is set
- Both honeycomb.apiKey or honeycomb.existingSecret are set
- honeycomb.existingSecret is set and honeycomb.existingSecretKey is not set

**Comment by TylerHelmuth** [comment]
@MikeGoldsmith we need to add a `ci` folder with a `default-values.yaml` file that sets `honeycomb.apiKey` to some bogus value. Here is what we do for the honeycomb kubernetes agent:

```
honeycomb:
  apiKey: example-api-key
```

File: charts/network-agent/templates/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "hny-network-agent.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
```
Comment (TylerHelmuth) [inline charts/network-agent/templates/daemonset.yaml:6]: We need to add these labels to all `metadata.labels` (only the labels at the root of the objects): https://github.com/open-telemetry/opentelemetry-helm-charts/blob/8402aeb044cf47f8f47590a3f094fdb79395a745/charts/opentelemetry-collector/templates/_helpers.tpl#L52-L62

File: charts/network-agent/templates/secret.yaml
Comment (TylerHelmuth) [inline charts/network-agent/templates/secret.yaml:1]: I would prefer we do not make a secret for users, instead depending on them to create a secret or inserting the API key directly.  I believe the work for this, and the warning, are back in the other PR.

Comment (MikeGoldsmith) [reply charts/network-agent/templates/secret.yaml:1]: I didn’t create the secret at first like in the other PR but found the user experience felt weird.

I think creating the secret to hold the API key if passed directly and include messaging to indicate the implications has a better user experience. Some security is better than none, right?

The readme, notes.txt and values.yaml all include details on what to expect. 

If we don’t want to manage the secret, I’d prefer to remove the option to pass in the API key directly and just expect a secret to be passed in all the time.

Comment (TylerHelmuth) [reply charts/network-agent/templates/secret.yaml:1]: I definitely see the value in the ergonomics of passing in the apiKey and making a secret for the user.  Another drawback, though, is that you either have to put the apiKey in the values.yaml file in plain text or in the install command via a `--set`. Although the method promotes security on the installed resources, it does not on the actual install steps.

With `--set` you can at least use and env var to hide the value, which is how we should endorse this functionality.

Comment (MikeGoldsmith) [reply charts/network-agent/templates/secret.yaml:1]: Yep, I agree setting using —set should use an env var. We do that in the agent readme. I’ll update the examples here to do the same. 

Maybe we suggest anyone using values.yaml to also use  secret to avoid putting their API key in plain text?

Are you okay keeping the secret if we do the example and recommendation above?

Comment (TylerHelmuth) [reply charts/network-agent/templates/secret.yaml:1]: Yes

File: charts/network-agent/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v2
+name: hny-network-agent
+description: Honeycomb Network Agent
+version: 0.0.1
+appVersion: v0.0.17-alpha
```
Comment (JamieDanielson) [inline charts/network-agent/Chart.yaml:5]:
```suggestion
appVersion: v0.0.19-alpha
```

File: charts/network-agent/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Honeycomb Network Agent
+
+The Helm chart install the [Honeycomb Network Agent](https://github.com/honeycombio/honeycomb-network-agent) in a Kubernetes cluster.
+
+## TL;DR;
+
+```bash
```
Comment (JamieDanielson) [inline charts/network-agent/README.md:7]:
```suggestion
```console
```

I would usually use shell for all of these, but it looks like console is the one consistently used the rest of this file and in the honeycomb readme at least.

File: charts/network-agent/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Honeycomb Network Agent
+
+The Helm chart install the [Honeycomb Network Agent](https://github.com/honeycombio/honeycomb-network-agent) in a Kubernetes cluster.
+
+## TL;DR;
```
Comment (JamieDanielson) [inline charts/network-agent/README.md:5]:
```suggestion
## TL;DR
```

**Review by JamieDanielson — COMMENTED** [review]
This looks good! A few suggestions on the first pass-through with the easy happy path of setting api key directly, confirmed the unpublished version here can be installed with the following:

```sh
export HONEYCOMB_API_KEY=<mykey>
helm install hny-network-agent --set honeycomb.apiKey=$HONEYCOMB_API_KEY ./charts/network-agent
```

File: charts/network-agent/templates/daemonset.yaml (hunk 2 above)
Comment (MikeGoldsmith) [reply charts/network-agent/templates/daemonset.yaml:6]: Added helpers to manage common and selector labels in https://github.com/honeycombio/helm-charts/pull/292/commits/c322ed778a57a87443e2a0d2c3ffbb962602a895

**Comment by MikeGoldsmith** [comment]
I think this ready for another (final?) review, thanks for your feedback 👍🏻 

cc @TylerHelmuth @JamieDanielson

File: charts/network-agent/values.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+honeycomb:
```
Comment (JamieDanielson) [inline charts/network-agent/values.yaml:1]: Do we want to add the debug and log level options or are we intentionally omitting them for the first pass?

Comment (MikeGoldsmith) [reply charts/network-agent/values.yaml:1]: I was going to say omit for now, but they definitely add value and will likely be more useful during initial feedback. I'll add them 👍🏻

File: charts/network-agent/templates/daemonset.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            - name: HONEYCOMB_API_ENDPOINT
+              value: {{ .Values.honeycomb.endpoint }}
+            - name: HONEYCOMB_DATASET
+              value: {{ .Values.honeycomb.dataset }}
+            - name: HONEYCOMB_STATS_DATASET
+              value: {{ .Values.honeycomb.statsDataset }}
+          securityContext:
+            capabilities:
+              add:
+                - NET_RAW
```
Comment (TylerHelmuth) [inline charts/network-agent/templates/daemonset.yaml:43]: Not a blocker for this PR, but It would be a good idea to add a section in the README detailing the permissions/RBAC this chart sets up.

**Review by TylerHelmuth — APPROVED** [review]
Left a couple nits, but nothing worth blocking anymore.

File: charts/network-agent/templates/daemonset.yaml (hunk 7 above)
Comment (MikeGoldsmith) [reply charts/network-agent/templates/daemonset.yaml:43]: We don't have a section in the main repo that describes required permissions yet. We've got the following ticket to track document them, and I've added a note to say we should update the helm chart once we've got it
- https://github.com/honeycombio/honeycomb-network-agent/issues/167

File: charts/network-agent/values.yaml (hunk 6 above)
Comment (MikeGoldsmith) [reply charts/network-agent/values.yaml:1]: I've added a simpler solution of allow extra env vars to be set and passed into the daemonset in https://github.com/honeycombio/helm-charts/pull/292/commits/c0545bde0c1debfec2c2d02e75a687453af14e17. Our and the OTel charts do the same thing.

I like this better because it doesn't let some accidentally to mess around with debug / development settings. They need to know what to set.
